src/components/ConnectButton/index.jsx

```diff
@@ -1,68 +1,111 @@
-import Fortmatic from 'fortmatic'
-import Portis from '@portis/web3'
-import Authereum from 'authereum'
+import Onboard from 'bnc-onboard'
 import Button from '~/components/layout/Button'
-import { fetchProvider, removeProvider } from '~/logic/wallets/store/actions'
-import { getNetwork } from '~/config'
+import { fetchProvider } from '~/logic/wallets/store/actions'
+import transactionDataCheck from '~/logic/wallets/transactionDataCheck'
+import { setWeb3, getWeb3 } from '~/logic/wallets/getWeb3'
+import { getNetworkId } from '~/config'
 import { store } from '~/store'
 
 const isMainnet = process.env.REACT_APP_NETWORK === 'mainnet'
 
-const PORTIS_DAPP_ID = isMainnet ? process.env.REACT_APP_PORTIS_ID : '852b763d-f28b-4463-80cb-846d7ec5806b'
+const BLOCKNATIVE_API_KEY = isMainnet
+  ? process.env.REACT_APP_BLOCKNATIVE_KEY
+  : '7fbb9cee-7e97-4436-8770-8b29a9a8814c'
+const PORTIS_DAPP_ID = isMainnet
+  ? process.env.REACT_APP_PORTIS_ID
+  : '852b763d-f28b-4463-80cb-846d7ec5806b'
 // const SQUARELINK_CLIENT_ID = isMainnet ? process.env.REACT_APP_SQUARELINK_ID : '46ce08fe50913cfa1b78'
-const FORTMATIC_API_KEY = isMainnet ? process.env.REACT_APP_FORTMATIC_KEY : 'pk_test_CAD437AA29BE0A40'
+const FORTMATIC_API_KEY = isMainnet
+  ? process.env.REACT_APP_FORTMATIC_KEY
+  : 'pk_test_CAD437AA29BE0A40'
 
-export const web3Connect = new Web3Connect.Core({
-  network: getNetwork().toLowerCase(),
-  providerOptions: {
-    walletconnect: {
-      package: WalletConnectProvider,
-      options: {
-        infuraId: process.env.REACT_APP_INFURA_TOKEN,
-      },
-    },
-    portis: {
-      package: Portis,
-      options: {
-        id: PORTIS_DAPP_ID,
-      },
-    },
-    fortmatic: {
-      package: Fortmatic,
-      options: {
-        key: FORTMATIC_API_KEY,
-      },
-    },
-    torus: {
-      package: Torus,
-      options: {
-        enableLogging: false,
-        buttonPosition: 'bottom-left',
-        buildEnv: process.env.NODE_ENV,
-        showTorusButton: true,
-      },
+const wallets = [
+  { walletName: 'metamask', preferred: true },
+  {
+    walletName: 'walletConnect',
+    preferred: true,
+    infuraKey: process.env.REACT_APP_INFURA_TOKEN,
+  },
+  {
+    walletName: 'trezor',
+    appUrl: 'gnosis-safe.io',
+    preferred: true,
+    email: 'safe@gnosis.io',
+    rpcUrl: 'https://rinkeby.infura.io/v3/b42c928da8fd4c1f90374b18aa9ac6ba',
+  },
+  {
+    walletName: 'ledger',
+    preferred: true,
+    rpcUrl: 'https://rinkeby.infura.io/v3/b42c928da8fd4c1f90374b18aa9ac6ba',
+  },
+  { walletName: 'trust', preferred: true },
+  { walletName: 'dapper' },
+  {
+    walletName: 'fortmatic',
+    apiKey: FORTMATIC_API_KEY,
+  },
+  {
+    walletName: 'portis',
+    apiKey: PORTIS_DAPP_ID,
+    label: 'Login with Email',
+  },
+  { walletName: 'authereum' },
+  { walletName: 'coinbase' },
+  { walletName: 'opera' },
+  { walletName: 'operaTouch' },
+]
+
+let lastUsedAddress = ''
+let providerName
```

Comment:
why do we need a variable to save provider name? As I can see it is only used once when passing to `fetchProvider` action

Reply:
Yes, but it is used by wallet and address callbacks as well

Reply:
ok now I see it, sorry :)